OpenKeychain/src/main/java/org/bouncycastle/openpgp/operator/jcajce/JcaPGPRawDigestContentVerifierBuilderProvider.java

```diff
@@ -0,0 +1,153 @@
+        {
+            this.keyAlgorithm = keyAlgorithm;
+            this.hashAlgorithm = hashAlgorithm;
+        }
+
+        public PGPContentVerifier build(final PGPPublicKey publicKey)
+                throws PGPException
+        {
+            final Signature signature = helper.createSignature(keyAlgorithm);
+
+            outputStream = new ByteArrayOutputStream();
+
+            try
+            {
+                signature.initVerify(keyConverter.getPublicKey(publicKey));
+            }
+            catch (InvalidKeyException e)
+            {
+                throw new PGPException("invalid key.", e);
+            }
+
+            return new PGPContentVerifier()
+            {
+                public int getHashAlgorithm()
+                {
+                    return hashAlgorithm;
+                }
+
+                public int getKeyAlgorithm()
+                {
+                    return keyAlgorithm;
+                }
+
+                public long getKeyID()
+                {
+                    return publicKey.getKeyID();
+                }
+
+                public boolean verify(byte[] expected)
+                {
+                    try {
+                        byte[] rawBytes = outputStream.toByteArray();
+                        byte[] encodedBytes = applyDigestEncoding(keyAlgorithm, hashAlgorithm, rawBytes);
+
+                        signature.update(encodedBytes);
+
+                        return signature.verify(expected);
+                    }
+                    catch (SignatureException e)
+                    {
+                        throw new PGPRuntimeOperationException("unable to verify signature: " + e.getMessage(), e);
+                    }
+                    catch (IOException e)
+                    {
+                        throw new IllegalStateException(e);
+                    }
+                }
+
+                public OutputStream getOutputStream()
+                {
+                    return outputStream;
+                }
+            };
+        }
+    }
+
+    private static byte[] applyDigestEncoding(int keyAlgorithm, int hashAlgorithm, byte[] bytes) throws IOException {
+        switch (keyAlgorithm) {
+            case PublicKeyAlgorithmTags.RSA_GENERAL:
+            case PublicKeyAlgorithmTags.RSA_SIGN:
+                try {
+                    String digestName = PGPUtil.getDigestName(hashAlgorithm);
+                    ASN1ObjectIdentifier digestIdentifier = DigestFactory.getOID(digestName);
+                    DigestInfo dInfo = new DigestInfo(
+                            new AlgorithmIdentifier(digestIdentifier, DERNull.INSTANCE), bytes);
+
+                    return dInfo.getEncoded(ASN1Encoding.DER);
+                } catch (PGPException e) {
+                    throw new IOException("Unknown hashing algorithm!", e);
+                }
+
+            default:
+                return bytes;
```

Comment:
I don't know. A result variable with break would be four extra lines, do you think that would be more clear? I think I like this style better personally, but single point of return also has its benefits. No strong opinion either way.